src/Sulu/Bundle/MediaBundle/Media/ListRepresentationFactory/MediaListRepresentationFactory.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\Media\ListRepresentationFactory;
+
+use Sulu\Bundle\MediaBundle\Entity\CollectionRepositoryInterface;
+use Sulu\Bundle\MediaBundle\Media\FormatManager\FormatManagerInterface;
+use Sulu\Bundle\MediaBundle\Media\Manager\MediaManagerInterface;
+use Sulu\Component\Media\SystemCollections\SystemCollectionManagerInterface;
+use Sulu\Component\Rest\ListBuilder\Doctrine\DoctrineListBuilder;
+use Sulu\Component\Rest\ListBuilder\Doctrine\DoctrineListBuilderFactoryInterface;
+use Sulu\Component\Rest\ListBuilder\Doctrine\FieldDescriptor\DoctrineFieldDescriptor;
+use Sulu\Component\Rest\ListBuilder\ListBuilderInterface;
+use Sulu\Component\Rest\ListBuilder\ListRepresentation;
+use Sulu\Component\Rest\ListBuilder\Metadata\FieldDescriptorFactoryInterface;
+use Sulu\Component\Rest\RestHelperInterface;
+use Sulu\Component\Security\Authentication\UserInterface;
+use Sulu\Component\Security\Authorization\PermissionTypes;
+use Sulu\Component\Security\Authorization\SecurityCheckerInterface;
+
+class MediaListRepresentationFactory
```

Comment:
Not sure if I would build it this way... Don't you think it would make sense to split that into multiple classes? I could imagine that if we split it into a `MediaListRepresentationFactory` and a `MediaListBuilderFactory` it would be easier to understand what is both components are doing. And I am not sure if I would include a `getFieldDescriptors` method, that contains a one-liner. Feels a bit odd to me (and it is again a function we would not allow to remove anymore). We could even split it further to have a `MediaFieldDescriptorFactory`, but since it is only a function call being passed `media`, I am not sure that would be worth it. /cc @alexander-schranz WDYT? Since you've built most of the `MediaBundle` :slightly_smiling_face:

Reply:
Splitting make sense for me and remove the single line function.